app/models/resource_request.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class ResourceRequest < ApplicationRecord
+  enum status: { pending: 0, in_progress: 1, success: 2, failure: 3, cancelled: 4 }
+  enum job_type: {
```

Comment:
so one approach to enums like this that you see historically is to define the enum values on bit switches - 0, 1, 2, 4, etc. - so that a given client value can encode multiple enum values (for example, being able to indicate fulltext on office or video access jobs). It might not be relevant here, and it has other implications about the total set of values for the enum, but I thought I would mention it.